sql/src/main/java/io/crate/planner/Planner.java

```diff
@@ -467,20 +458,30 @@ public Plan visitSetStatement(SetAnalyzedStatement setStatement, PlannerContext
             case SESSION:
                 return new SetSessionPlan(setStatement.settings());
             case GLOBAL:
+                if (setStatement.isPersistent()) {
+                    return new UpdateSettingsPlan(setStatement.settings(), true);
+                } else {
+                    return new UpdateSettingsPlan(
+                        Collections.emptyMap(),
+                        setStatement.settings(),
+                        true
+                    );
+                }
             default:
                 if (setStatement.isPersistent()) {
-                    return new UpdateSettingsPlan(setStatement.settings());
+                    return new UpdateSettingsPlan(setStatement.settings(), false);
                 } else {
                     return new UpdateSettingsPlan(
                         Collections.emptyMap(),
-                        setStatement.settings()
+                        setStatement.settings(),
+                        false
```

Comment:
also here it looks the same like the ``GLOBAL`` impl, ``setStatement.scope() == GLOBAL`` could be used for the ``UpdateSettingsPlan``.